File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/eventdata/GameStartedFromSavedStateEventData.java
package com.kafkaquest.kq.common.util.models.events.eventdata;

import com.kafkaquest.kq.common.util.models.events.other.JigsawPieceCurrentPosition;
import com.kafkaquest.kq.common.util.models.events.other.JigsawPuzzlePiece;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameStartedFromSavedStateEventData extends EventData {

    private int puzzleImageId;

    private int horizontalSize;

    private int verticalSize;

    private long elapsedTime;

    private long penaltyTime;

    private List<JigsawPuzzlePiece> puzzlePieces;

    private List<JigsawPieceCurrentPosition> addedPieces;
}

File: kq-projection-dotnet/kq-projection/Common/Models/Events/GameStartedFromSavedStateEventData.cs
using KQ_Projection.Common.Models.Jigsaw;
using System.Text.Json.Serialization;

namespace KQ_Projection.Common.Models.Events
{
    public class GameStartedFromSavedStateEventData : EventData
    {
        [JsonPropertyName("puzzleImageId")]
        public int puzzleImageId {  get; set; }

        [JsonPropertyName("horizontalSize")]
        public int horizontalSize { get; set; }

        [JsonPropertyName("verticalSize")]
        public int verticalSize { get; set; }

        [JsonPropertyName("elapsedTime")]
        public long elapsedTime { get; set; }

        [JsonPropertyName("penaltyTime")]
        public long penaltyTime { get; set; }

        [JsonPropertyName("puzzlePieces")]
        public List<JigsawPuzzlePiece> puzzlePieces { get; set; }

        [JsonPropertyName("addedPieces")]
        public List<JigsawPieceCurrentPosition> addedPieces { get; set; }
    }
}
